src/widgets/auto-styler.js

```diff
@@ -0,0 +1,141 @@
+var cdb = require('cartodb.js');
+var CategoryColors = require('./category/category-colors');
+var AutoStyler = cdb.core.Model.extend({
+  initialize: function (dataviewModel) {
+    this.dataviewModel = dataviewModel;
+    this.colors = new CategoryColors();
+    this.vector = !!this.dataviewModel._dataProvider;
```

Comment:
Again, this class shouldn't know about how the map is rendered... That was the goal of deep-insights.js (separating widgets from map rendering). In this class we have several conditionals that make it work differently for raster and vector... CartoDB.js's API should provide methods so that we don't need to things like accessing `this.dataviewModel._dataProvider`...